core/trino-main/src/main/java/io/trino/operator/FlatHashStrategyCompiler.java

```diff
@@ -466,7 +466,7 @@ private static MethodDefinition generateHashBlockVectorized(ClassDefinition defi
             rleHandling.append(new ForLoop("for (int index = 0; index < length; index++) { hashes[index] = CombineHashFunction.getHash(hashes[index], hash); }")
                     .initialize(index.set(constantInt(0)))
                     .condition(lessThan(index, length))
-                    .update(index.increment())
+                    .update(new BytecodeBlock().incrementVariable(index, (byte) 1))
```

Comment:
I assume this means using IINC, actually matters (which seems strange given how HotSpot works), so can you update Variable and ByteCodeExpression instead, so all code gets this benefit?

Reply:
It doesn't _really_ matter, but it is smaller in terms of bytecode size- `IINC` only works for `byte / short / int` but not `long` whereas the sequence of `variable = variable + 1` works regardless of size (but involves more bytecodes). I actually did this because I noticed that this was how you implemented the loop increments in `AccumulatorCompiler` which I assumed was on purpose on your behalf. I'll drop it from this PR and open the `IINC` logic against airlift.

Reply:
airlift/bytecode PR: https://github.com/airlift/bytecode/pull/17